rasa/shared/utils/io.py

```diff
@@ -264,8 +264,9 @@ def get_list_fingerprint(
     Returns:
         the fingerprint of the list
     """
+    ordered_elements = sorted(elements, key=lambda x: str(x))
```

Comment:
I am not convinced sorting the list is a good idea. This fingerprinting is used in a number of places, and a list inherently is a data structure where you'd expect the order of items to play a role. E.g. let's say we fingerprint the pipeline configuration, the order of components plays a big role. 

Can we rather explicitly use an unordered list in the places where we do not care about the order and only apply sorting to them?